src/huggingface_hub/inference/_providers/hf_inference.py

```
@@ -151,6 +151,14 @@ def _check_supported_task(model: str, task: str) -> None:
             return  # Only conversational allowed if tagged as conversational
         raise ValueError("Non-conversational image-text-to-text task is not supported.")
 
+    if (
+        task in ("feature-extraction", "sentence-similarity")
+        and pipeline_tag in ("feature-extraction", "sentence-similarity")
+        and task in tags
+    ):
+        # feature-extraction and sentence-similarity are interchangeable for HF Inference
+        return
```

👍 

it's the same code as in JS, right?

---

In JS we don't have a check on the task. We should be able to implement it once https://github.com/huggingface/huggingface.js/pull/1315 is merged cc @hanouticelina

---

ah yes but i only meant this code:
```ts
override makeRoute(params: UrlParams): string {
	if (params.task && ["feature-extraction", "sentence-similarity"].includes(params.task)) {
		// when deployed on hf-inference, those two tasks are automatically compatible with one another.
		return `pipeline/${params.task}/${params.model}`;
	}
	return `models/${params.model}`;
}
```

---

(not sure if we need to check the task, i haven't seen any issues)

---

the check is not necessary, it was mainly to "fail early" and have a more comprehensive error message

---

> (not sure if we need to check the task, i haven't seen any issues)

added in https://github.com/huggingface/huggingface_hub/pull/2948#issue-2937770453 by @hanouticelina :

_Currently, when users call a model with an unsupported task, the request silently times out without any helpful error message._

---

The errors are actually a bit more diverse than just timeouts, for example you can get a `Template error: template not found` error if you call a text-generation model that does not have a chat template (no `conversational` tag). these errors are not that common since most users (i hope) follow the inference snippets on the model page.

---

(i was saying this for JS, not necessarily for Python) 

Yes so this requires us to do 1 extra http call, but i understand it can improve the DX a bit, so i have no strong opinion :)

---

> Yes so this requires us to do 1 extra http call,

Note that the call is cached so only the first prediction has an extra http call for HF Inference.
In the end it's the same as for other providers for which we do an extra http call to fetch the provider mapping.

(and yeah, neglectable compared to improved DX)